vscode-trace-extension/src/extension.ts

```diff
@@ -102,3 +124,38 @@ export function activate(context: vscode.ExtensionContext): void {
 export function deactivate(): void {
     traceLogger.disposeChannel();
 }
+
+async function startTraceServerIfAvailable(): Promise<void>{
+    const traceServerExtension = vscode.extensions.getExtension('tracecompass-community.vscode-trace-server');
+    if (!traceServerExtension || await isUp()) {
+        pendingOpen = undefined;
+        return;
+    }
+    const extensionId = 'vscode-trace-server';
+    vscode.commands.executeCommand(extensionId + '.start-if-stopped');
```

Comment:
In the context of `openTrace` below, it doesn't seem guaranteed that `pendingOpen` will either be consumed or be set to `undefined`. If the command is executed but neither returned nor awaited, then in `openTrace`, we don't get into the `catch` block that would set `pendingOpen` to `undefined`, but it isn't guaranteed that the other plugin will eventually call `serverStatus.started` (e.g. if the server fails to start), so `pendingOpen` would remain set to the old value.

Reply:
With my planed updates this won't be necessary anymore.